### PR TITLE
Updated Interaction System

### PR DESCRIPTION
**Pull request by DevulTj**
- Instead of pressing E to show the interaction menu, all you have to do is hover over an entity.
- You use your scroll wheel to scroll through possible interactions
- Then you press E to attempt to resolve an interaction

It's sexy.

### CODE REVIEW

File: code/UI/Interactions/InteractionMenu.cs [hunk 1]
```diff
@@ -28,65 +61,95 @@ public void SetEntity( Entity entity )
...
+			Clear();
 
+			return;
 		}
+
+		if ( input.MouseWheel != 0f )
+			DoMouseWheelInput( input.MouseWheel );
+
+		if ( input.Pressed( InputButton.Use ) )
+			Use();
```
Comment (trundler-dev) [inline code/UI/Interactions/InteractionMenu.cs:153]: Pressing or holding "Use" after you select an InteractionEntry is going to result in an ArgumentOutOfRange Exception here.

File: code/UI/Interactions/InteractionMenu.cs [hunk 2]
```diff
@@ -28,65 +61,95 @@ public void SetEntity( Entity entity )
...
-		{
-			interaction.ClientResolve();
+		UpdateArrows();
+	}
 
-			if ( interaction.ResolveOnServer )
-				Interaction.TryServerResolve( interaction.Owner.NetworkIdent, interaction.ID );
+	protected void Use()
+	{
+		var interaction = InteractionList[CurrentInteractionIndex];
```
Comment (trundler-dev) [inline code/UI/Interactions/InteractionMenu.cs:94]: To follow up on above, after selecting an InteractionEntry, we clear the list, and are still trying to access index n.

**Review by trundlr — CHANGES_REQUESTED** [review]

File: code/UI/Interactions/InteractionMenu.cs (hunk 1 above)
Comment (DevulTj) [reply code/UI/Interactions/InteractionMenu.cs:153]: I'll fix these tomorrow

**Review by trundlr — APPROVED** [review]
Approved, once you fix that small issue :)

**Pull request merged**